.gitignore

```diff
@@ -2,3 +2,6 @@ node_modules
 build
 .sizecache.json
 *.log*
+# Eclipse project files
+/.settings/
+/.project
```

Comment:
Hey I also use eclipse and love it ^^. In this case there's no need to store the IDE specific files in the source repository, it is discouraged since it adds unnecessary verbosity for those that do not use eclipse. You can ignore the eclipse's dot files [globally](https://help.github.com/articles/ignoring-files#create-a-global-gitignore), in this case you can work in the project without worring about unmodified files being marked to be staged by EGit.